src/Controller/ProfileController.php

```diff
@@ -67,4 +68,21 @@ public function edit(
             'formpassword' => $formpassword
         ]);
     }
+
+    /**
+     * @Route("/profile/delete", name="profile_delete")
+     */
+    public function delete(EntityManagerInterface $entityManager, Request $request): Response
+    {
+        $user = $this->getUser();
+        $entityManager = $this->getDoctrine()->getManager();
+        if ($user != null) {
+            $entityManager->remove($user);
+            $entityManager->flush();
+            $session = new Session();
```

Comment:
You should try to get the Session from the Request object (and get the Request by injecting it to the controller) instead of creating a new one.